### DISCUSSION TITLE
Allow primery constructors on all partial delcarations as long as they exactly match

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#7998] [Language Ideas]
Currently modifiers and attribute, base list are merged, and generic constraints must match if present but primary constructors use a difference limitation. I think primary constructors could be consistent with generic constraints on this point, because it is sometimes useful to have either of them repeated in every part.

**Comment by CyrusNajmabadi** [comment]
> because it is sometimes useful to have either of them repeated in every part.

When is that case?  Thanks!

**Reply by alrz** [reply]
This could be considered a code style preference but sometimes there's no "main" type part so you may choose to repeat the generic constraints to be visible when editing either file or just to be explicit when you spell out the "default" just like `private` modifier. The same reasoning directly applies to partial types with primary constructors.
